src/main.rs

```diff
@@ -96,8 +129,159 @@ fn main() {
                 port,
             );
 
-            // TODO: create client instance
+            // create client instance and connect to server
+            let mut connection = TcpStream::connect((ip, port)).unwrap();
+
+            // add CTRL+C system hook, so that connection partner is informed about disconnect
+            let client_conn_clone = connection.try_clone().unwrap();
+            let running = Arc::new(AtomicBool::new(true));
+            let _ = running.clone();
+            ctrlc::set_handler(move || {
+                net::send(&mut client_conn_clone.try_clone().unwrap(), MessageType::Quit);
+                client_conn_clone.shutdown(Shutdown::Both).expect("shutdown call failed");
+            }).expect("Error setting Ctrl+C handler");
+
+            let mut board = ::model::types::Board::init();
+            if let Some(b) = client_args.value_of("board") {
+                board = util::read_extern_board(b);
+            }
+
+            let mut client = ::model::types::Player {
+                own_board: board.clone(),
```

Comment:
Oh, I need the board to calculate the ships on it for the capacity. Did not see a possibility to access `own_board` in the struct initialization, so I cloned the board...